src/pytest_plugins/filler/witness.py

```diff
@@ -0,0 +1,119 @@
+"""
+Pytest plugin for witness functionality.
+
+Provides --witness command-line option that installs the witness-filler tool
+and generates execution witness data for blockchain test fixtures when enabled.
+"""
+
+import subprocess
+from typing import Any, Callable
+
+import pytest
+
+from ethereum_test_fixtures.blockchain import FixtureBlock, WitnessChunk
+
+
+def pytest_addoption(parser: pytest.Parser):
+    """Add witness command-line options to pytest."""
+    witness_group = parser.getgroup("witness", "Arguments for witness functionality")
+    witness_group.addoption(
+        "--witness",
+        "--witness-the-fitness",
+        action="store_true",
+        dest="witness",
+        default=False,
+        help=(
+            "Install the witness-filler tool and generate execution witness data for blockchain "
+            "test fixtures."
+        ),
+    )
+
+
+def pytest_configure(config):
+    """
+    Pytest hook called after command line options have been parsed.
+
+    If --witness is enabled, installs the witness-filler tool from the specified
+    git repository.
+    """
+    if config.getoption("witness"):
+        print("🔧 Installing witness-filler tool from kevaundray/reth...")
+        print("   This may take several minutes for first-time compilation...")
+
+        result = subprocess.run(
+            [
+                "cargo",
+                "install",
+                "--git",
+                "https://github.com/kevaundray/reth.git",
+                "--rev",
+                "8016a8a5736e4427b3d285c82cd39c4ece70f8c4",
+                "witness-filler",
+            ],
+        )
+
+        if result.returncode != 0:
+            pytest.exit(
+                f"Failed to install witness-filler tool (exit code: {result.returncode}). "
+                "Please ensure you have a compatible Rust toolchain installed. "
+                "You may need to update your Rust version to 1.86+ or run without --witness.",
+                1,
+            )
+        else:
+            print("✅ witness-filler tool installed successfully!")
+
+
+@pytest.fixture
+def witness_generator(request: pytest.FixtureRequest) -> Callable[[Any], None] | None:
+    """
+    Provide a witness generator function if --witness is enabled.
+
+    Returns:
+        None if witness functionality is disabled.
+        Callable that generates witness data for a fixture if enabled.
+
+    """
+    if not request.config.getoption("witness"):
+        return None
+
+    def generate_witness(fixture: Any) -> None:
+        """Generate witness data for a fixture using the witness-filler tool."""
+        if not hasattr(fixture, "blocks") or not fixture.blocks:
+            return
```

Comment:
We should instead constrain the type of parameter `fixture` to be a specific fixture type, because:
- Using `hasattr` too many times
- `witness-filler`, whatever it does, it's probably only going to support one of our fixture types
- The caller of the function is the one that should use `isinstance` before calling this function instead of trying to guess what were we passed as a parameter.